packages/stream_chat_flutter/lib/src/message_input/stream_message_input.dart

```diff
@@ -549,11 +551,31 @@ class StreamMessageInputState extends State<StreamMessageInput>
       // correctly in the UI.
       _onChangedDebounced.call();
 
+      final channel = StreamChannel.of(context).channel;
+      final config = StreamChatConfiguration.of(context);
+
       // Resumes the cooldown if the channel has currently an active cooldown.
       if (!_isEditing) {
-        final channel = StreamChannel.of(context).channel;
         _effectiveController.startCooldown(channel.getRemainingCooldown());
       }
+
+      if (config.draftMessagesEnabled) {
+        // Starts listening to the draft stream for the current channel/thread.
+        final draftStream = switch (_effectiveController.message.parentId) {
+          final parentId? => channel.state?.threadDraftStream(parentId),
+          _ => channel.state?.draftStream,
+        };
+
+        _draftStreamSubscription = draftStream?.distinct().listen((draft) {
+          // If the draft is removed, reset the controller.
+          if (draft == null) return _effectiveController.reset();
+
+          // Otherwise, update the controller with the draft message.
+          if (draft.message case final draftMessage) {
+            _effectiveController.message = draftMessage.toMessage();
+          }
+        });
+      }
```

Comment:
I would extract this to a separate method, the initState is already very long

Reply:
Done